src/content/blog/astro-570.mdx

```diff
@@ -0,0 +1,235 @@
+
+- **[Node](https://docs.astro.build/en/guides/integrations-guide/node/#sessions)**: zero-config filesystem driver
+- **[Netlify](https://docs.astro.build/en/guides/integrations-guide/netlify/#sessions)**: zero-config Netlify Blobs driver
+- **[Cloudflare](https://docs.astro.build/en/guides/integrations-guide/cloudflare/#sessions)**: minimal-config Workers KV driver
+- **[Vercel](https://docs.astro.build/en/guides/integrations-guide/vercel/#sessions)**: minimal-config Redis/Upstash driver
+
+On any of these platforms you can easily switch out the driver for a different one if you don't want to use the default.
+
+```js title="astro.config.mjs" {5-10}
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  // ...
+  session: {
+    driver: 'redis',
+    options: {
+      url: process.env.REDIS_URL,
+    },
+  },
+});
+
+```
+
+### Upgrading from experimental sessions
+
+If you were using the experimental sessions API, you should remove the experimental flag from your `astro.config.mjs` file:
+
+```js title="astro.config.mjs" del={6}
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  // ...
+  experimental: {
+    session: false,
+  },
+});
+```
+Check out the [sessions guide](https://docs.astro.build/en/guides/sessions/) to learn more about how to use them in your Astro project.
+
+## SVG Components
+
+...
+
+## Config Imports
+
+The new virtual module `astro:config` is now stable!
+
+The virtual module allows to import a subset of the Astro configuration, with type-safety in mind. Two sub-paths are exposed for controlled access to your configuration:
+
+- `astro:config/client`: exposes config information that is safe to expose to the client.
+- `astro:config/server`: exposes additional information that is safe to expose to the server, such as file/dir paths.
+
+For example, you can now build your own path utilities without relying on an integrations:
+
+```js title="astro.config.mjs"
+// astro.config.mjs
+export default defineConfig({
+    base: "store/",
+    trailingSlash: "always"
+})
+```
+
+```js title="src/utils.js"
+// src/utils.js
+import { trailingSlash, base } from 'astro:config/client';
+
+// This method adds a forwards slash to paths, and prepends the `base`
+function addForwardSlashAndBase(path) {
+    let finalPath;
+    if (trailingSlash === 'always') {
+      finalPath = path.endsWith('/') ? path : path + '/';
+    } else {
+      finalPath = path;
+    }
+    if (base !== "/") {
+        return base + finalPath
+    }
+    return finalPath
+}
+```
+
+```astro title="src/pages/index.astro"
+---
+// src/pages/index.astro
+import { addForwardSlashAndBase } from "@src/utils.js";
+
+const cartUrl = addForwardSlashAndBase("cart"); // it will return "store/cart/"
+---
+
+<a href={cartUrl}>Cart</a>
+```
+
```

Comment:
```suggestion

We `config`ured you'd enjoy this new ability to access your settings, with Astro's trademark type safety to help you get it right!
```
OK, doesn't HAVE to be a jokey sentence, but I think it might be nice to close this off with *something* about this being a requested feature, or we hope you enjoy it etc. Especially given it's the last entry, and it finished off with a lot of code samples. A conclusion of some sort would be nice here, I think.